airflow/www/views.py

```diff
@@ -1082,6 +1088,54 @@ def datasets(self):
             state_color_mapping=state_color_mapping,
         )
 
+    @expose("/dataset_dags/<int:dataset_id>")
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DATASET),
+        ]
+    )
+    @action_logging
+    def retrieve_related_dags_for_dataset(self, dataset_id: int):
+        """
+        Retrieve related DAGs for a specific dataset.
+
+        :param dataset_id: The dataset id
+        """
+        with create_session() as session:
+            dataset = session.query(DatasetModel).filter_by(id=dataset_id).one_or_none()
+            if not dataset:
+                return flask.json.jsonify({"error": "Dataset not found"}), 404
+
+            # Fetch all consuming DAGs (downstream)
+            consuming_dags_refs = (
+                session.query(DagScheduleDatasetReference).filter_by(dataset_id=dataset_id).all()
+            )
+            consuming_dags = [
+                {
+                    "dag_id": dag_ref.dag_id,
+                    "is_paused": session.query(DagModel).filter_by(dag_id=dag_ref.dag_id).one().is_paused,
```

Comment:
Wondering if we can optimise the number of DB calls here.
Instead of making a query every single time when iterating over consuming_dag_refs, can we get pre-build the list of all the dag_ids from consuming_dag_regs and make a single query to DagModel with all those DAG IDs for getting there is_paused values?

Reply:
Added the change to reduce the DB calls: https://github.com/apache/airflow/pull/38369/commits/838d92f228ee0083135d331610dff59234db9357